inline_test.go

```diff
@@ -570,13 +570,16 @@ func TestInlineLink(t *testing.T) {
 
 		// no closing ), autolinking detects the url next
 		"[disambiguation](http://en.wikipedia.org/wiki/Disambiguation_(disambiguation) is the",
-		"<p>[disambiguation](<a href=\"http://en.wikipedia.org/wiki/Disambiguation_(disambiguation\">http://en.wikipedia.org/wiki/Disambiguation_(disambiguation</a>) is the</p>\n",
+		"<p>[disambiguation](<a href=\"http://en.wikipedia.org/wiki/Disambiguation_(disambiguation)\">http://en.wikipedia.org/wiki/Disambiguation_(disambiguation)</a> is the</p>\n",
```

Comment:
It changes the way unbalanced parenthesis is handled in links. This results in forming a correct link by giving priority to the parenthesis in the link. This however, does not conform(like earlier) to [common mark processing](http://spec.commonmark.org/dingus/?text=foo%20%5Bdisambiguation%5D(http%3A%2F%2Fen.wikipedia.org%2Fwiki%2FDisambiguation_(disambiguation)%20is%20the).

I was trying to make it follow the common mark spec but I could get my way around the code. 🤔